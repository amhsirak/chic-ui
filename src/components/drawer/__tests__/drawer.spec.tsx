import ReactDOM from 'react-dom';
import Drawer from '../drawer';
import { render } from '../../../test/utils';

/* eslint-disable */
describe('Drawer Component', () => {
  it('Renders Correctly', () => {
    ReactDOM.createPortal = jest.fn((children, container) => {
      return children;
    }) as any;

    const DrawerGroup = render(
      <>
        <Drawer placement="right" header="Drawer Header" footer="Drawer Footer">
          <div>Some contents...</div>
          <div> Some contents...</div>
          <div> Some contents...</div>
        </Drawer>
        <Drawer
          placement="left"
          closable={false}
          header="Drawer Header"
          footer="Drawer Footer"
        >
          <div>Some contents...</div>
          <div> Some contents...</div>
          <div> Some contents...</div>
        </Drawer>
        <Drawer placement="top">
          <div>Some contents...</div>
          <div> Some contents...</div>
          <div> Some contents...</div>
        </Drawer>
        <Drawer placement="bottom" size="large" closeIcon="X">
          <div>Some contents...</div>
          <div> Some contents...</div>
          <div> Some contents...</div>
        </Drawer>
      </>
    );
    expect(DrawerGroup).toMatchSnapshot();
  });
});
