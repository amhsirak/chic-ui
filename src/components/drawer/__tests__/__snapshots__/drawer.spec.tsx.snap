// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Drawer Component Renders Correctly 1`] = `
Array [
  <div
    className="sc-bdfBQB fYOROy"
    onClick={[Function]}
    open={false}
    tabIndex={-1}
  >
    <div
      className="sc-gsTEea gCzLUm"
      height="100%"
      onClick={[Function]}
      open={false}
      width="378px"
    >
      <div
        className="sc-dlfnuX kxBbVN"
      >
        <button
          aria-label="close"
          className="sc-hKgJUU haPkAg"
          onClick={[Function]}
        >
          <svg
            viewBox="0 0 490 490"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M456.851 0L245 212.564 33.149 0 .708 32.337l211.961 212.667L.708 457.678 33.149 490 245 277.443 456.851 490l32.441-32.322-211.961-212.674L489.292 32.337z"
            />
          </svg>
        </button>
        Drawer Header
      </div>
      <div
        className="sc-eCstlR dDYssu"
      >
        <div>
          Some contents...
        </div>
        <div>
           Some contents...
        </div>
        <div>
           Some contents...
        </div>
      </div>
      <div
        className="sc-jSgvzq WblV"
      >
        Drawer Footer
      </div>
    </div>
  </div>,
  <div
    className="sc-bdfBQB fYOROy"
    onClick={[Function]}
    open={false}
    tabIndex={-1}
  >
    <div
      className="sc-gsTEea gFHkhO"
      height="100%"
      onClick={[Function]}
      open={false}
      width="378px"
    >
      <div
        className="sc-dlfnuX kxBbVN"
      >
        Drawer Header
      </div>
      <div
        className="sc-eCstlR dDYssu"
      >
        <div>
          Some contents...
        </div>
        <div>
           Some contents...
        </div>
        <div>
           Some contents...
        </div>
      </div>
      <div
        className="sc-jSgvzq WblV"
      >
        Drawer Footer
      </div>
    </div>
  </div>,
  <div
    className="sc-bdfBQB fYOROy"
    onClick={[Function]}
    open={false}
    tabIndex={-1}
  >
    <div
      className="sc-gsTEea igEWIm"
      height="378px"
      onClick={[Function]}
      open={false}
      width="100%"
    >
      <div
        className="sc-dlfnuX kxBbVN"
      >
        <button
          aria-label="close"
          className="sc-hKgJUU haPkAg"
          onClick={[Function]}
        >
          <svg
            viewBox="0 0 490 490"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M456.851 0L245 212.564 33.149 0 .708 32.337l211.961 212.667L.708 457.678 33.149 490 245 277.443 456.851 490l32.441-32.322-211.961-212.674L489.292 32.337z"
            />
          </svg>
        </button>
        ­
      </div>
      <div
        className="sc-eCstlR dDYssu"
      >
        <div>
          Some contents...
        </div>
        <div>
           Some contents...
        </div>
        <div>
           Some contents...
        </div>
      </div>
    </div>
  </div>,
  <div
    className="sc-bdfBQB fYOROy"
    onClick={[Function]}
    open={false}
    tabIndex={-1}
  >
    <div
      className="sc-gsTEea cSeZmA"
      height="736px"
      onClick={[Function]}
      open={false}
      width="100%"
    >
      <div
        className="sc-dlfnuX kxBbVN"
      >
        <button
          aria-label="close"
          className="sc-hKgJUU haPkAg"
          onClick={[Function]}
        >
          X
        </button>
        ­
      </div>
      <div
        className="sc-eCstlR dDYssu"
      >
        <div>
          Some contents...
        </div>
        <div>
           Some contents...
        </div>
        <div>
           Some contents...
        </div>
      </div>
    </div>
  </div>,
]
`;
